sqlmesh/dbt/model.py

```diff
@@ -156,23 +156,56 @@ def model_kind(self, target: TargetConfig) -> ModelKind:
             return ModelKind(name=ModelKindName.EMBEDDED)
         raise ConfigError(f"{materialization.value} materialization not supported.")
 
-    @property
-    def sql_no_config(self) -> str:
-        matches = re.findall(r"{{\s*config\(", self.sql)
-        if matches:
-            config_macro_start = self.sql.index(matches[0])
-            cursor = config_macro_start
+    def _sql_extract_config(self) -> t.Tuple[SqlStr, SqlStr]:
+        def match_end(str: str, start: int) -> int:
+            cursor = start
             quote = None
-            while cursor < len(self.sql):
-                if self.sql[cursor] in ('"', "'"):
+            while cursor < len(sql_no_config):
+                if str[cursor] in ('"', "'"):
                     if quote is None:
-                        quote = self.sql[cursor]
-                    elif quote == self.sql[cursor]:
+                        quote = str[cursor]
+                    elif quote == str[cursor]:
                         quote = None
-                if self.sql[cursor : cursor + 2] == "}}" and quote is None:
-                    return "".join([self.sql[:config_macro_start], self.sql[cursor + 2 :]])
+                if str[cursor : cursor + 2] == "}}" and quote is None:
+                    return cursor + 2
                 cursor += 1
-        return self.sql
+            return cursor
+
+        sql_no_config: str = self.sql
+        only_config = ""
+
+        matches = re.findall(r"{{\s*config\s*\(", sql_no_config)
+        for match in matches:
+            start = sql_no_config.find(match)
```

Comment:
I don't understand. I only want the first occurrence.

Reply:
Ah, ignore me, sorry. I just read the code again and saw that you replace the extracted config string with the empty string, so this should be fine since every time `find` will visit all config blocks, right?